recipe/meta.yaml

```diff
@@ -26,7 +29,9 @@ requirements:
     - cross-python_{{ target_platform }}     # [build_platform != target_platform]
     - {{ posix }}bison
     - {{ posix }}flex
-    - meson
+    # use last working meson to get PyPy build to work
+    # see https://github.com/mesonbuild/meson/issues/8570
+    - meson 0.55.*
```

Comment:
This is not a sustainable workaround IMHO, especially with the additional patches necessary to get this working.

I am not convinced that getting gobject-introspection to work with something as exotic as Pypy is worth putting in a constraint like this.

Reply:
That's definitely a valid concern. I have no idea about how often the GLib/GNOME/Gtk+ libraries are used with PyPy, except that I can say that I've never seen a request or issue about the setup except from the bot.

And the hope is that a Meson version pin will only be temporary because we'll get the underlying issue fixed:

https://github.com/mesonbuild/meson/issues/8570

But in my experience it can take a while for that to happen, and PyPy in particular does not seem to be a high priority for the Meson team, so it's a definite worry.

I wonder if we could alter the migrator to not pursue PyPy migrations for feedstocks in this section of the dependency graph? We've put in a lot of work to make the migrations work as best we can and to be honest I'm sure if there's really been much of a point to it. And it's much harder to distribute the tacit knowledge of "oh just ignore this category of PRs" than it is to avoid creating the PRs in the first place.

As for this particular PR, I guess it would be more conservative to not add PyPy support at all, than to add it and potentially take it a way later. So maybe it would be best to determine if there's any actual demand for this support before committing to it.

Reply:
I can just about live with meson pinning being brought in when it is necessary to ensure builds for macOS Silicon or Windows to function properly, but it seems too much for something like PyPy IMHO.

Reply:
I do have curiosity, if not quite interest, in getting PyPy to work for my pet package `gnuradio` which is a couple dependencies down from this in the graph. It could turn out to be very interesting, but I won't know until I can try it.

I also think conda-forge is in an interesting place on the frontier of supporting PyPy, but any potential adoption is likely to be slow or non-existent if half the package graph remains unported. That's not really `gobject-introspection`'s fault, but it is a contributor. It's not that people would necessarily demand that the GNOME/GTK family support PyPy, but I think it's important for the ecosystem to add support where possible.

Now, in lieu of getting movement upstream to fix the meson issue, perhaps we would be better served by patching conda-forge's meson than pinning to an older version. @pkgw you point out fixing an 'if' statement in that bug report, which would be an easy patch to justify. But you also mention a further issue with macOS. Do you think coming up with a patch there would be feasible? I can take a look if so.

Reply:
@ryanvolz Based on my read of the code I *think* that negating the one boolean ought to fix both PyPy and macOS. If the proverbial "someone" were to put the effort into investigating such a patch, I do bet that it would help spur upstream action.

Reply:
Sounds good, I'll look into it.

Reply:
It would be great to add PyPy as a test dependency for Meson -- currently we have a CI test that checks if python modules can be built for various versions of python, including PyPy, but it is skipped because PyPy is not installed. @mattip is currently working on that in conjunction with fixes for PyPy support on Windows.